demes/demes.py

```diff
@@ -1257,6 +1258,7 @@ class Graph:
             iterable_validator=attr.validators.instance_of(list),
         ),
     )
+    metadata: dict = attr.ib(factory=dict, validator=attr.validators.instance_of(dict))
```

Comment:
This looks quite strict, wouldn't ``instanceof(collections.abc.Mapping)`` be enough?